app/models/katello/glue/candlepin/pool.rb

```diff
@@ -191,7 +191,9 @@ def hosts
       end
 
       def create_activation_key_associations
-        keys = Resources::Candlepin::ActivationKey.get(nil, "?include=id&include=pools.pool.id", organization.label)
+        keys = Rails.cache.fetch("#{organization.label}/activation_keys", expires_in: 2.minutes) do
```

Comment:
I would probably name this something like:
"#{organization.label}/activation_keys_id_pool_id"